Sources/StreamChat/StateLayer/Chat.swift

```diff
@@ -168,6 +178,89 @@ public final class Chat {
         try await channelUpdater.removeMembers(currentUserId: currentUserId, cid: cid, userIds: Set(members), message: systemMessage)
     }
     
+    /// Loads an array of members for the specified query.
+    ///
+    /// - Parameters:
+    ///   - filter: The filter to use for fetching channel members.
+    ///   - sort: An array of sorting parameters.
+    ///   - pagination: The pagination configuration which includes a limit and an offset or a cursor.
+    ///
+    /// - Throws: An error while communicating with the Stream API.
+    /// - Returns: An array of paginated channel members for the query.
+    public func loadMembers(with filter: Filter<MemberListFilterScope>, sort: [Sorting<ChannelMemberListSortingKey>] = [], pagination: Pagination) async throws -> [ChatChannelMember] {
+        var query = ChannelMemberListQuery(cid: cid, filter: filter, sort: sort)
+        query.pagination = pagination
+        return try await memberListUpdater.load(query)
+    }
```

Comment:
I think this one makes sense to be here in the Chat object, and maybe for this one, we can follow what @martinmitrevski said, and have a `members` property in the state.

Reply:
Thanks, I'll try out a hybrid approach where `ChatState` has members and there is still a way to create member lists without `Chat`.

Reply:
It is implemented now